src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -112,28 +168,43 @@ abstract class LazyModule()(implicit val p: Parameters)
     children.filter(!_.omitGraphML).foreach { c => c.edgesGraphML(buf, pad) }
   }
 
+  /** execute function to [[children]].
+    * @param iterfunc function to run
+    * */
   def nodeIterator(iterfunc: (LazyModule) => Unit): Unit = {
     iterfunc(this)
     children.foreach( _.nodeIterator(iterfunc) )
   }
 
+  /** get [[children]] */
   def getChildren = children
 }
 
 object LazyModule
 {
+  /** current [[LazyModule]] scope, specifically, it is a stack of [[LazyModule]].
+    * it will be dynamically set by [[LazyScope.apply]] and [[LazyModule.apply]],
+    * */
   protected[diplomacy] var scope: Option[LazyModule] = None
+  /** index of [[LazyModule]], notice there is no No.0 module. */
   private var index = 0
 
+  /** add a [[LazyModule]]
+    * @param bc [[LazyModule]] needed to be instantiated.
```

Comment:
I don't get what `bc` is. This note doesn't help me. What does `bc` stand for...? @terpstra ?

Reply:
do you get what `bc` is now?